packages/react/src/Delay.spec.tsx

```diff
@@ -16,9 +24,25 @@ describe('<Delay/>', () => {
     render(<Delay>{TEXT}</Delay>)
     expect(screen.queryByText(TEXT)).toBeInTheDocument()
   })
-  it('should accept 0 for ms prop', async () => {
+  it('should accept 0 for ms prop', () => {
     render(<Delay ms={0}>{TEXT}</Delay>)
-    await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument(), { timeout: 1000 })
+
+    expect(screen.queryByText(TEXT)).toBeInTheDocument()
+  })
+  it('should render fallback content initially and then the actual text after the delay', async () => {
+    render(
+      <Delay ms={ms('1s')} fallback={<p role="paragraph">fallback</p>}>
+        {TEXT}
+      </Delay>
+    )
+
+    vi.advanceTimersByTime(ms('0.5s'))
+
+    expect(screen.queryByRole('paragraph')).toBeInTheDocument()
+
+    vi.advanceTimersByTime(ms('0.5s'))
+
+    await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument())
```

Comment:
How about strictly by timeout option of waitFor?

```suggestion
    await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument(), {
      timeout: ms('1s'),
    })
```

Reply:
@manudeli 
I know that option doesn't make much sense if we use a `fake timer`, but do you think it would be better to add it?  

```tsx
  it('should render fallback content initially and then the actual text after the delay', async () => {
    render(/* ... */)
    // vi.advanceTimersByTime(ms('0.5s'))

    expect(screen.queryByRole('paragraph')).toBeInTheDocument()

    // vi.advanceTimersByTime(ms('0.5s'))

    await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument(), { timeout: 5000 })
  })
```
In the above case, the timeout of waitFor is valid. 

```tsx
  it('should render fallback content initially and then the actual text after the delay', async () => {
    render(/* ... */)
    // vi.advanceTimersByTime(ms('0.5s'))

    expect(screen.queryByRole('paragraph')).toBeInTheDocument()

    // vi.advanceTimersByTime(ms('0.5s'))

    await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument()) // 
  })
```

However, if we don't use fake time as shown above and don't specify the timeout option, the test will fail because, as you said, the default value of timeout is 1000ms.

```tsx
  it('should render fallback content initially and then the actual text after the delay', async () => {
    render(/* ... */)

    vi.advanceTimersByTime(ms('0.5s'))

    expect(screen.queryByRole('paragraph')).toBeInTheDocument()

    vi.advanceTimersByTime(ms('0.5s'))

    await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument())
  })
```
If we use a fake timer, we don't necessarily need the timeout option - the above test will work fine.

Reply:
@manudeli Um.. I couldn't get an accurate test for timeout when I used `ms()`, so I entered it as a number.

Reply:
I made the time value large for testing purposes. 🙏

Reply:
https://stackoverflow.com/questions/65970116/react-testing-library-waitfor-not-working-with-settimeout/65970771#65970771?newreg=dcb8efd321ab40cea4e0e0646d9b6934